Packs/AnythingLLM/Integrations/AnythingLLM/AnythingLLM.py

```diff
@@ -257,15 +260,77 @@ def workspace_delete(self, workspace:str ):
 
         return {"message": {"success": True, "message": "Workspace removed successfully"}}
 
+    def workspace_thread_new(self, workspace: str, thread: str):
+        try:
+            wslug = workspace_slug(workspace, self.workspace_list())
+            try:
+                exists = False
+                thread_slug(workspace, self.thread_list(workspace))
+                exists = True
+            except Exception:
+                # enforce unique thread names and always append "_slug" to the name
+                # to create the slug
+                data = {
+                    'name': thread,
+                    'slug': thread + "_slug"
+                }
+                response = self._http_request(
+                    method="POST",
+                    url_suffix=f"/v1/workspace/{wslug}/thread/new",
+                    json_data=data
+                )
+                return response
+            finally:
+                if exists:  # pylint: disable=E0601
+                    raise Exception("workspace thread already exists")
+        except Exception as e:
+            msg = f"AnythingLLM: workspace_thread_new: exception creating a new workspace [{workspace}] - {str(e)}"
+            demisto.debug(msg)
+            raise Exception(msg)
```

Comment:
There's a lot of nested exceptions which may over complicate the flow. You can try something like this and it would achieve what you're looking for.
```python
def workspace_thread_new(self, workspace: str, thread: str):
    try:
        wslug = workspace_slug(workspace, self.workspace_list())
        if not wslug:
            raise Exception(f"Workspace '{workspace}' does not exist.")

        threads = self.thread_list(workspace)

        if any(t['name'] == thread for t in threads):
            raise Exception(f"Thread '{thread}' already exists in workspace '{workspace}'.")

        data = {
            'name': thread,
            'slug': f"{thread}_slug"
        }

        response = self._http_request(
            method="POST",
            url_suffix=f"/v1/workspace/{wslug}/thread/new",
            json_data=data
        )
        return response

    except Exception as e:
        msg = f"Error creating new thread '{thread}' in workspace '{workspace}': {str(e)}"
        demisto.debug(msg)
        raise Exception(msg)
```

Reply:
Implemented this basic approach